indexing-service/src/main/java/org/apache/druid/indexing/overlord/ForkingTaskRunner.java

```diff
@@ -477,6 +429,8 @@ public TaskStatus call()
                         portFinder.markPortUnused(tlsChildPort);
                       }
 
+                      getTracker().returnStorageSlot(storageSlot);
```

Comment:
shouldn't you delete the directories first before returning the slot? Similar to how it's happening in BaseRestorableTaskRunner. In that class, if deleting directory fails for any reason, the slot will not get unreserved.